src/ai/defaultai.cc

```diff
@@ -1844,158 +1844,95 @@ void DefaultAI::update_buildable_field(BuildableField& field) {
 	field.unconnected_nearby = false;
 
 	// collect information about productionsites nearby
-	static std::vector<Widelands::ImmovableFound> immovables;
-	immovables.reserve(50);
-	immovables.clear();
-	// Search in a radius of range
-	map.find_immovables(
-	   game(), Widelands::Area<Widelands::FCoords>(field.coords, kProductionArea + 2), &immovables);
-
-	// function seems to return duplicates, so we will use serial numbers to filter them out
-	static std::set<uint32_t> unique_serials;
-	unique_serials.clear();
-
-	for (const Widelands::ImmovableFound& imm_found : immovables) {
-		const Widelands::BaseImmovable& base_immovable = *imm_found.object;
-		if (!unique_serials.insert(base_immovable.serial()).second) {
-			continue;  // serial was not inserted in the set, so this is a duplicate
-		}
-
-		if (upcast(Widelands::PlayerImmovable const, player_immovable, &base_immovable)) {
-
-			// TODO(unknown): Only continue if this is an opposing site
-			// allied sites should be counted for military influence
-			if (player_immovable->owner().player_number() != pn) {
-				continue;
-			}
-			// here we identify the buiding (including expected building if constructionsite)
-			// and calculate some statistics about nearby buildings
-			if (player_immovable->descr().type() == Widelands::MapObjectType::PRODUCTIONSITE) {
-				BuildingObserver& bo = get_building_observer(player_immovable->descr().name().c_str());
-				consider_productionsite_influence(field, imm_found.coords, bo);
-			} else if (upcast(Widelands::ConstructionSite const, constructionsite, player_immovable)) {
-				const Widelands::BuildingDescr& target_descr = constructionsite->building();
-				if (target_descr.type() == Widelands::MapObjectType::PRODUCTIONSITE) {
-					BuildingObserver& bo = get_building_observer(target_descr.name().c_str());
-					consider_productionsite_influence(field, imm_found.coords, bo);
-				}
-			}
-		}
-	}
-
-	// Now testing military aspects
-	immovables.clear();
-	map.find_immovables(game(),
-	                    Widelands::Area<Widelands::FCoords>(field.coords, actual_enemy_check_area),
-	                    &immovables);
 
 	// We are interested in unconnected immovables, but we must be also close to connected ones
 	static bool any_connected_imm = false;
```

Comment:
this variable is only used to determine field.unconnected_nearby which is not used anywhere after it has been set so it should be safe to delete both variables
see https://github.com/widelands/widelands/pull/5013/commits/1dfda555f8d26f8016cce459b2da0982e647be1e

Reply:
Good catch. However as we already have this info, I will try to use it in genetic algorithm. I put a todo there. Probably in my next branch